de-lib/src/main/java/org/iplantc/de/apps/client/presenter/tools/ManageToolsViewPresenter.java

```diff
@@ -0,0 +1,54 @@
+package org.iplantc.de.apps.client.presenter.tools;
+
+import org.iplantc.de.apps.client.views.ManageToolsView;
+import org.iplantc.de.client.gin.ServicesInjector;
+import org.iplantc.de.client.models.tool.Tool;
+import org.iplantc.de.client.services.ToolServices;
+import org.iplantc.de.commons.client.ErrorHandler;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+import com.google.inject.Inject;
+
+import java.util.List;
+
+/**
+ * Created by sriram on 4/24/17.
+ */
+public class ManageToolsViewPresenter implements ManageToolsView.Presenter {
+
+    @Inject
+    ManageToolsView toolsView;
+
+    ManageToolsView.ManageToolsViewAppearance appearance;
+
+    ToolServices dcService = ServicesInjector.INSTANCE.getDeployedComponentServices();
+    
+    @Inject
+    public ManageToolsViewPresenter(ManageToolsView.ManageToolsViewAppearance appearance) {
+        this.appearance = appearance;
+    }
+
+    @Override
+    public void go(HasOneWidget container) {
+       container.setWidget(toolsView.asWidget());
+       toolsView.mask(appearance.mask());
+       loadTools();
+    }
+
+    @Override
+    public void loadTools() {
+            dcService.getTools(new AsyncCallback<List<Tool>>() {
```

Comment:
Also, all of the Tools service calls should be `AppsCallback` instead of `AsyncCallback`.  This way, if the apps service is down, the Apps window, the App Launch window, and now this new Tools window will be disabled together.  That will also require the Tools window to be set up with a `WindowType` and config through the desktop window manager and whatnot like I mentioned earlier though.